nemo/tron/state.py

```diff
@@ -204,10 +184,39 @@ def fault_tolerance_state(self, value: FaultToleranceState):
 
     @property
     def signal_handler(self):
+        if self._signal_handler is None:
+            self._set_signal_handler()
```

Comment:
This needs to be called at `__init__` because the property gets called too late (in `checkpoint_and_decide_exit`

Reply:
`_set_signal_handler` is also called in the `cfg.setter` above. this was the earliest point i could see where we could initialize the handler outside of the `__init__`. it's not the most obvious though line though, and it does rely on the config being the first thing that is set on the global state.